dotnet/src/Connectors/Connectors.MistralAI/Client/MistralClient.cs

```diff
@@ -133,12 +134,8 @@ internal async Task<IReadOnlyList<ChatMessageContent>> GetChatMessageContentsAsy
 
             Debug.Assert(kernel is not null);
 
-            // Add the original assistant message to the chatRequest; this is required for the service
-            // to understand the tool call responses. Also add the result message to the caller's chat
-            // history: if they don't want it, they can remove it, but this makes the data available,
-            // including metadata like usage.
-            chatRequest.AddMessage(chatChoice.Message!);
-
+            // Add the result message to the caller's chat history;
+            // this is required for the service to understand the tool call responses.
             var chatMessageContent = this.ToChatMessageContent(modelId, responseData, chatChoice);
             chatHistory.Add(chatMessageContent);
```

Comment:
For streaming this is a single line statement, can we do this?

`chatHistory.Add(this.ToChatMessageContent(modelId, responseData, chatChoice));`

Reply:
This one can be upgraded to `static` as well.

Reply:
I wanted to do exactly the same, but the `chatMessageContent` variable is used in line 174 below for the auto function invocation filter's context:
<img width="1035" alt="image" src="https://github.com/user-attachments/assets/89661e89-eb9f-43f2-a94d-12205c8df756">